### PR TITLE
Bug Fix: Cards mapping

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/mappings/signals.py [hunk 1]
```diff
@@ -29,10 +27,6 @@ def run_post_mapping_settings_triggers(sender, instance: MappingSetting, **kwarg
     if instance.is_custom:
         schedule_fyle_attributes_creation(int(instance.workspace_id))
 
-    workspace_general_settings = WorkspaceGeneralSettings.objects.filter(workspace_id=instance.workspace_id).first()
```
Comment (Sravanksk) [inline apps/mappings/signals.py:32]: At this point we won't have WorkspaceGeneralSettings for new orgs, so this is failing and not required

File: apps/workspaces/utils.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def create_or_update_general_settings(general_settings_payload: Dict, workspace_
 
 
 def delete_cards_mapping_settings(workspace_general_settings: WorkspaceGeneralSettings):
-    if not workspace_general_settings.map_fyle_cards_qbo_account:
+    if not workspace_general_settings.map_fyle_cards_qbo_account or not workspace_general_settings.corporate_credit_card_expenses_object:
```
Comment (ashwin1111) [inline apps/workspaces/utils.py:172]: Why do we even post this mapping setting when ccc exports are not enabled in the first place?

File: apps/workspaces/models.py [hunk 3]
```diff
@@ -83,6 +83,7 @@ class WorkspaceGeneralSettings(models.Model):
     category_sync_version = models.CharField(default='v1', max_length=50, help_text='Category sync version')
     je_single_credit_line = models.BooleanField(default=False, help_text='Single Credit Line for Journal Entries')
     map_fyle_cards_qbo_account = models.BooleanField(default=True, help_text='Map Fyle Cards to QBO Accounts')
+    skip_cards_mapping = models.BooleanField(default=False, help_text='Skip cards mapping')
```
Comment (ashwin1111) [inline apps/workspaces/models.py:86]: Can you write a script to set this as true for onboarded orgs

File: apps/mappings/signals.py [hunk 4]
```diff
@@ -29,10 +27,6 @@ def run_post_mapping_settings_triggers(sender, instance: MappingSetting, **kwarg
...
-
-    delete_cards_mapping_settings(workspace_general_settings)
```
Comment (ashwin1111) [inline apps/mappings/signals.py:34]: why is this removed?

File: apps/workspaces/utils.py (hunk 2 above)
Comment (Sravanksk) [reply apps/workspaces/utils.py:172]: This will be enabled for all orgs by default, so just to ignore one edge case

Comment (ashwin1111) [reply apps/workspaces/utils.py:172]: Why can't you exclude them? We are inserting via sql query right?

File: apps/mappings/signals.py (hunk 4 above)
Comment (Sravanksk) [reply apps/mappings/signals.py:34]: this will be called post mapping_settings, For new orgs at this point we don't have workspace_general_settings
This call is still in progress IMO, so this is throwing 500

Comment (ashwin1111) [reply apps/mappings/signals.py:34]: hey, why not handle them with a if condition? if POST APIs time differ then we will definitely won't be consistent since we are calling both apis in parallel from app.

File: apps/mappings/signals.py (hunk 1 above)
Comment (ashwin1111) [reply apps/mappings/signals.py:32]: safely ignore them

File: apps/workspaces/utils.py (hunk 2 above)
Comment (Sravanksk) [reply apps/workspaces/utils.py:172]: Yes, this condition will come incase config update, just to catch that case, have added this check

**Review by ashwin1111 — APPROVED** [review]
